compiler/test/dotty/tools/dotc/util/SparseIntArrayTests.scala

```diff
@@ -0,0 +1,38 @@
+package dotty.tools.dotc.util
+
+import org.junit.Assert._
+import org.junit.Test
+
+class SparseIntArrayTests:
+  @Test
+  def sparseArrayTests: Unit =
```

Comment:
I tried this out and it can only handle keys up to `2^30 - 1`, or else there is an infinite loop